fastcrypto-zkp/src/conversions.rs

```diff
@@ -0,0 +1,461 @@
+    out
+}
+
+pub fn blst_fr_to_bls_fr(fe: blst_fr) -> BlsFr {
+    let mut out = [0u64; 4];
+    unsafe { blst_uint64_from_fr(out.as_mut_ptr(), &fe) };
+    let bytes = out.as_byte_slice();
+
+    BlsFr::from_le_bytes_mod_order(bytes)
+}
+
+// Base Field conversions
+pub fn bls_fq_to_blst_fp(f: Fq) -> blst_fp {
+    let mut fp_bytes_le = [0u8; G1_UNCOMPRESSED_SIZE / 2];
+    f.serialize_uncompressed(&mut fp_bytes_le[..])
+        .expect("fp size correct");
+
+    let mut blst_fp = blst_fp::default();
+    unsafe {
+        blst_fp_from_lendian(&mut blst_fp, fp_bytes_le.as_ptr());
+    }
+    blst_fp
+}
+
+pub fn blst_fp_to_bls_fq(f: blst_fp) -> Fq {
+    let mut out = [0u64; 6];
+    unsafe { blst_uint64_from_fp(out.as_mut_ptr(), &f) };
+    let bytes = out.as_byte_slice();
+    <Fq as FromBytes>::read(bytes).unwrap()
+}
+
+// QFE conversions
+pub fn bls_fq2_to_blst_fp2(f: Fq2) -> blst_fp2 {
+    let mut fp_bytes_le = [0u8; G2_UNCOMPRESSED_SIZE / 2];
+    f.serialize_uncompressed(&mut fp_bytes_le[..])
+        .expect("fp size correct");
+
+    blst_fp2 {
+        fp: fp_bytes_le
+            .chunks(48)
+            .map(|fp_bytes| {
+                let mut blst_fp = blst_fp::default();
+                unsafe {
+                    blst_fp_from_lendian(&mut blst_fp, fp_bytes.as_ptr());
+                }
+                blst_fp
+            })
+            .collect::<Vec<blst_fp>>()
+            .try_into()
+            .unwrap(),
+    }
+}
+
+pub fn blst_fp2_to_bls_fq2(f: blst_fp2) -> Fq2 {
+    let [fp1, fp2] = f.fp;
+    let bls_fp1 = blst_fp_to_bls_fq(fp1);
+    let bls_fp2 = blst_fp_to_bls_fq(fp2);
+    Fq2::new(bls_fp1, bls_fp2)
+}
+
+// Target Field conversions
+pub fn bls_fq6_to_blst_fp6(f: Fq6) -> blst_fp6 {
+    let c0 = bls_fq2_to_blst_fp2(f.c0);
+    let c1 = bls_fq2_to_blst_fp2(f.c1);
+    let c2 = bls_fq2_to_blst_fp2(f.c2);
+    blst_fp6 { fp2: [c0, c1, c2] }
+}
+
+pub fn blst_fp6_to_bls_fq6(f: blst_fp6) -> Fq6 {
+    let c0 = blst_fp2_to_bls_fq2(f.fp2[0]);
+    let c1 = blst_fp2_to_bls_fq2(f.fp2[1]);
+    let c2 = blst_fp2_to_bls_fq2(f.fp2[2]);
+    Fq6::new(c0, c1, c2)
+}
+
+pub fn bls_fq12_to_blst_fp12(f: Fq12) -> blst_fp12 {
+    let c0 = bls_fq6_to_blst_fp6(f.c0);
+    let c1 = bls_fq6_to_blst_fp6(f.c1);
+    blst_fp12 { fp6: [c0, c1] }
+}
+
+pub fn blst_fp12_to_bls_fq12(f: blst_fp12) -> Fq12 {
+    let c0 = blst_fp6_to_bls_fq6(f.fp6[0]);
+    let c1 = blst_fp6_to_bls_fq6(f.fp6[1]);
+    Fq12::new(c0, c1)
+}
+
+/// Affine point translations: those mostly allow us to receive the
+/// proof points, provided in affine form.
+pub fn bls_g1_affine_to_blst_g1_affine(pt: &BlsG1Affine) -> blst_p1_affine {
+    debug_assert_eq!(pt.uncompressed_size(), G1_UNCOMPRESSED_SIZE);
+    let tmp_p1 = blst_p1_affine {
+        x: bls_fq_to_blst_fp(pt.x),
+        y: bls_fq_to_blst_fp(pt.y),
+    };
+    // See https://github.com/arkworks-rs/curves/issues/14 for why the double serialize
```

Comment:
+1

Reply:
It would be great to have a a bulletpoint list of what we offer to the community after this epic. BLST backend, better timings, fixes / extra safety over existing applications (correct masks etc), better alignment with standards etc. This can be part of a readme subsection re importance and promising results of our implementation. It can be a follow up PR.